### PR TITLE
quic: fix compiler errors and warnings

### PR DESCRIPTION
**Pull request by danbev**
This commit fixes the reported compiler errors and warning when building
with `--experimental-quic`.

<details>
<summary>errors/warnings:</summary>

```console
../src/node_quic_session.h:67:15: error: inline function 'node::quic::QuicSessionConfig::ResetToDefaults' is not defined [-Werror,-Wundefined-inline]
  inline void ResetToDefaults();
              ^
../src/node_quic_session.h:45:5: note: used here
    ResetToDefaults();
    ^
../src/node_quic_session.h:73:15: error: inline function 'node::quic::QuicSessionConfig::Set' is not defined [-Werror,-Wundefined-inline]
  inline void Set(
              ^
../src/node_quic_session.h:50:5: note: used here
    Set(env);
    ^
../src/node_quic_session.h:554:21: error: inline function 'node::quic::QuicSession::OnReceiveClientInitial' is not defined [-Werror,-Wundefined-inline]
  static inline int OnReceiveClientInitial(
                    ^
../src/node_quic_session.h:1129:5: note: used here
    OnReceiveClientInitial,
    ^
../src/node_quic_session.h:558:21: error: inline function 'node::quic::QuicSession::OnReceiveCryptoData' is not defined [-Werror,-Wundefined-inline]
  static inline int OnReceiveCryptoData(
                    ^
../src/node_quic_session.h:1130:5: note: used here
    OnReceiveCryptoData,
    ^
../src/node_quic_session.h:565:21: error: inline function 'node::quic::QuicSession::OnHandshakeCompleted' is not defined [-Werror,-Wundefined-inline]
  static inline int OnHandshakeCompleted(
                    ^
../src/node_quic_session.h:1131:5: note: used here
    OnHandshakeCompleted,
    ^
../src/node_quic_session.h:568:25: error: inline function 'node::quic::QuicSession::OnDoHSEncrypt' is not defined [-Werror,-Wundefined-inline]
  static inline ssize_t OnDoHSEncrypt(
                        ^
../src/node_quic_session.h:1133:5: note: used here
    OnDoHSEncrypt,
    ^
../src/node_quic_session.h:581:25: error: inline function 'node::quic::QuicSession::OnDoHSDecrypt' is not defined [-Werror,-Wundefined-inline]
  static inline ssize_t OnDoHSDecrypt(
                        ^
../src/node_quic_session.h:1134:5: note: used here
    OnDoHSDecrypt,
    ^
../src/node_quic_session.h:594:25: error: inline function 'node::quic::QuicSession::OnDoEncrypt' is not defined [-Werror,-Wundefined-inline]
  static inline ssize_t OnDoEncrypt(
                        ^
../src/node_quic_session.h:1135:5: note: used here
    OnDoEncrypt,
    ^
../src/node_quic_session.h:607:25: error: inline function 'node::quic::QuicSession::OnDoDecrypt' is not defined [-Werror,-Wundefined-inline]
  static inline ssize_t OnDoDecrypt(
                        ^
../src/node_quic_session.h:1136:5: note: used here
    OnDoDecrypt,
    ^
../src/node_quic_session.h:620:25: error: inline function 'node::quic::QuicSession::OnDoInHPMask' is not defined [-Werror,-Wundefined-inline]
  static inline ssize_t OnDoInHPMask(
                        ^
../src/node_quic_session.h:1137:5: note: used here
    OnDoInHPMask,
    ^
../src/node_quic_session.h:629:25: error: inline function 'node::quic::QuicSession::OnDoHPMask' is not defined [-Werror,-Wundefined-inline]
  static inline ssize_t OnDoHPMask(
                        ^
../src/node_quic_session.h:1138:5: note: used here
    OnDoHPMask,
    ^
../src/node_quic_session.h:638:21: error: inline function 'node::quic::QuicSession::OnReceiveStreamData' is not defined [-Werror,-Wundefined-inline]
  static inline int OnReceiveStreamData(
                    ^
../src/node_quic_session.h:1139:5: note: used here
    OnReceiveStreamData,
    ^
../src/node_quic_session.h:652:21: error: inline function 'node::quic::QuicSession::OnAckedCryptoOffset' is not defined [-Werror,-Wundefined-inline]
  static inline int OnAckedCryptoOffset(
                    ^
../src/node_quic_session.h:1140:5: note: used here
    OnAckedCryptoOffset,
    ^
../src/node_quic_session.h:658:21: error: inline function 'node::quic::QuicSession::OnAckedStreamDataOffset' is not defined [-Werror,-Wundefined-inline]
  static inline int OnAckedStreamDataOffset(
                    ^
../src/node_quic_session.h:1141:5: note: used here
    OnAckedStreamDataOffset,
    ^
../src/node_quic_session.h:676:21: error: inline function 'node::quic::QuicSession::OnStreamOpen' is not defined [-Werror,-Wundefined-inline]
  static inline int OnStreamOpen(
                    ^
../src/node_quic_session.h:1142:5: note: used here
    OnStreamOpen,
    ^
../src/node_quic_session.h:670:21: error: inline function 'node::quic::QuicSession::OnStreamClose' is not defined [-Werror,-Wundefined-inline]
  static inline int OnStreamClose(
                    ^
../src/node_quic_session.h:1143:5: note: used here
    OnStreamClose,
    ^
../src/node_quic_session.h:734:21: error: inline function 'node::quic::QuicSession::OnStatelessReset' is not defined [-Werror,-Wundefined-inline]
  static inline int OnStatelessReset(
                    ^
../src/node_quic_session.h:1144:5: note: used here
    OnStatelessReset,
    ^
../src/node_quic_session.h:687:21: error: inline function 'node::quic::QuicSession::OnRand' is not defined [-Werror,-Wundefined-inline]
  static inline int OnRand(
                    ^
../src/node_quic_session.h:1148:5: note: used here
    OnRand,
    ^
../src/node_quic_session.h:693:21: error: inline function 'node::quic::QuicSession::OnGetNewConnectionID' is not defined [-Werror,-Wundefined-inline]
  static inline int OnGetNewConnectionID(
                    ^
../src/node_quic_session.h:1149:5: note: used here
    OnGetNewConnectionID,
    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[1]: *** [/Users/danielbevenius/work/nodejs/quic/node_quic/out/Release/obj.target/libnode/src/node_quic.o] Error 1
make[1]: *** Waiting for unfinished jobs....
../src/node_quic_crypto.cc:1002:10: warning: unused variable 'pos' [-Wunused-variable]
  size_t pos = 0;
         ^
In file included from ../src/node_quic_crypto.cc:1:
In file included from ../src/node_quic_crypto.h:6:
In file included from ../src/node_crypto.h:30:
In file included from ../src/env.h:27:
In file included from ../src/aliased_buffer.h:7:
../src/util.h:264:43: error: inline function 'node::ContainerOf<uv_timer_s, node::quic::Timer>' is not defined [-Werror,-Wundefined-inline]
constexpr ContainerOfHelper<Inner, Outer> ContainerOf(Inner Outer::*field,
                                          ^
../src/node_quic_util.h:439:22: note: used here
          Timer* t = ContainerOf(
                     ^
1 warning and 1 error generated.
make[1]: *** [/Users/danielbevenius/work/nodejs/quic/node_quic/out/Release/obj.target/libnode/src/node_quic_crypto.o] Error 1
rm 23b9307565d9fe2fcb314bdb403fb6b4b5600dbc.intermediate dfebb38eeb918ef8c8ea45832c53e5452f56debf.intermediate cca4bbec20be3370a730595aba68a1a175696f85.intermediate
make: *** [node] Error 2
 ```
</details>

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: src/node_quic_util.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
 
 #include "node_quic_buffer.h"
+#include "util-inl.h"
```
Comment (addaleax) [inline src/node_quic_util.h]: What is this one needed for? It’s probably a good sign that something should go into a `node_quic_util-inl.h` or `node_quic_util.cc`.

(Like, this isn’t important, it’s just something that should be figured out in the long run)

Comment (danbev) [reply src/node_quic_util.h]: It was required to fix this error:
```
In file included from ../src/node_quic_crypto.cc:1:
In file included from ../src/node_quic_crypto.h:6:
In file included from ../src/node_crypto.h:30:
In file included from ../src/env.h:27:
In file included from ../src/aliased_buffer.h:7:
../src/util.h:264:43: error: inline function 'node::ContainerOf<uv_timer_s, node::quic::Timer>' is not defined [-Werror,-Wundefined-inline]
constexpr ContainerOfHelper<Inner, Outer> ContainerOf(Inner Outer::*field,
                                          ^
../src/node_quic_util.h:439:22: note: used here
          Timer* t = ContainerOf(
```
I'd be happy to take a closer look at this tomorrow.

Comment (addaleax) [reply src/node_quic_util.h]: Yeah, the `Timer` class methods shouldn’t be in this header :)

Comment (jasnell) [reply src/node_quic_util.h]: Yep, I was going to move those at some point :-)

**Comment by danbev** [comment]
Landed in 72997d659a84635db791fdf9d8a80d61b326ad8f.

**Pull request closed without merging**